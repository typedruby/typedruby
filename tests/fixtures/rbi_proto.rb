class A
  # Demonstrate that we'll use a .rbi prototype when typechecking an
  # invokee
  def untyped_with_stub(x)
    x + 1
  end

  def typed(String y) => String
    untyped_with_stub(y)
  end
end
