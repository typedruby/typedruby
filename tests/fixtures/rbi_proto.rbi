class A
  def untyped_with_stub(Integer x) => Integer
  end
end
