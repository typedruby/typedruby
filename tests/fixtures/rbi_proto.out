error: Could not match types:

        @ __ROOT__/tests/fixtures/rbi_proto.rbi:2
      2 |    def untyped_with_stub(Integer x) => Integer
                                   ^^^^^^^ Integer, with:
        @ __ROOT__/tests/fixtures/rbi_proto.rb:8
      8 |    def typed(String y) => String
                       ^^^^^^ String

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/rbi_proto.rbi:2
      2 |    def untyped_with_stub(Integer x) => Integer
                                   ^^^^^^^ Integer, with:
        @ __ROOT__/tests/fixtures/rbi_proto.rb:9
      9 |      untyped_with_stub(y)
                                 ^ String
        @ __ROOT__/tests/fixtures/rbi_proto.rb:9
      9 |      untyped_with_stub(y)
               ^^^^^^^^^^^^^^^^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/rbi_proto.rb:8
      8 |    def typed(String y) => String
                                    ^^^^^^ String, with:
        @ __ROOT__/tests/fixtures/rbi_proto.rbi:2
      2 |    def untyped_with_stub(Integer x) => Integer
                                                 ^^^^^^^ Integer

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/rbi_proto.rb:8
      8 |    def typed(String y) => String
                                    ^^^^^^ String, with:
        @ __ROOT__/tests/fixtures/rbi_proto.rb:9
      9 |      untyped_with_stub(y)
               ^^^^^^^^^^^^^^^^^^^^ Integer
